### PR TITLE
Feature: Creating a main makefile to encapsulate common build actions.

### PR DESCRIPTION
**Pull request by KevinJCross**

### CODE REVIEW

File: src/autoscaler/db/sqldb/lock_sqldb_test.go [hunk 1]
```diff
@@ -98,7 +98,7 @@ var _ = Describe("LockSqldb", func() {
 		Context("when the lock does not exist", func() {
 			Context("because the row does not exist", func() {
 				BeforeEach(func() {
-					lock = &models.Lock{Owner: "123456", Ttl: testTTL}
+					lock = createLock("123456", testTTL)
```
Comment (garethjevans) [inline src/autoscaler/db/sqldb/lock_sqldb_test.go:101]: I'm not sure these changes should be in here, they seem to have come from the other PR.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+target/init:
+	@make -C src/autoscaler buildtools
+	@touch $@
+
+.PHONY: clean
+clean:
+	@rm target/* &> /dev/null || echo "# Already clean"
+
+.PHONY: build-db
+build-db: target/build-db
```
Comment (garethjevans) [inline Makefile:62]: can we change the .github/workflows/mysql|postgres.yml to call these new targets to ensure they are tested

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	@[ $(shell cf plugins | grep AutoScaler | wc -l) -gt 1  ] || cf install-plugin -f -r CF-Community app-autoscaler-plugin
+	@touch $@
+target/install-cli-Ubuntu:
+	@echo " - not installing $@"
+target/install-cli-Linux:
+	@echo " - not installing $@"
+
+.PHONY: init
+init: target/init
+target/init:
```
Comment (garethjevans) [inline Makefile:53]: Will need some documentation updates on the main README.md

**Review by garethjevans — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (KevinJCross) [reply Makefile:53]: yup :) ... just seeing if everyone agrees before changing over.

File: src/autoscaler/db/sqldb/lock_sqldb_test.go (hunk 1 above)
Comment (KevinJCross) [reply src/autoscaler/db/sqldb/lock_sqldb_test.go:101]: yeah its a bit old Ill update.

Comment (KevinJCross) [reply src/autoscaler/db/sqldb/lock_sqldb_test.go]: out of this PR :)

File: Makefile (hunk 2 above)
Comment (KevinJCross) [reply Makefile:64]: done :)

File: Makefile (hunk 3 above)
Comment (KevinJCross) [reply Makefile:53]: done

**Review by garethjevans — APPROVED** [review]

**Review by bonzofenix — APPROVED** [review]
lgtm

**Pull request merged**